File: frontend/src/types/customCourse.ts
export type Lesson = {
  title: string
  content: string
  summary: string
}

export type Video = {
  title: string
  url: string
}

export type CustomCourse = {
  id?: number
  user_id: string
  title: string
  description?: string
  lessons: Lesson[]
  videos: Video[]
}

File: backend/models/custom_course.py
from typing import List, Optional
from pydantic import BaseModel

class LessonInput(BaseModel):
    title: str
    content: str

class VideoInput(BaseModel):
    title: str
    url: str

class CustomCourseInput(BaseModel):
    user_id: str
    title: str
    description: Optional[str]
    lessons: List[LessonInput]
    videos: List[VideoInput]

class LessonOutput(LessonInput):
    id: int

class VideoOutput(VideoInput):
    id: int

class CustomCourseOutput(BaseModel):
    id: int
    user_id: str
    title: str
    description: Optional[str]
    lessons: List[LessonOutput]
    videos: List[VideoOutput]
